docs/advanced_topics/performance.md

```diff
@@ -80,6 +80,47 @@ The methods used to get the URL of a `Page` such as `Page.get_url` and `Page.get
 
 When using the [`{% pageurl %}`](pageurl_tag) or [`{% fullpageurl %}`](fullpageurl_tag) template tags, the request is automatically passed in, so no further optimization is needed.
 
+## Deferred fields
+
+Some APIs require using the `Page` model, such as when you want to search across all pages on your site. However, you need to then render or call methods on the specific instances.
+
+These specific instances can be retrieved using the `specific` attribute on either the page or queryset:
+
+```python
+>>> Page.objects.all().specific()
+[<BlogPage: A Blog post>, <BlogPage: Another Blog post>]
+>>> Page.objects.first()
+<Page: A Blog post>
+>>> Page.objects.first().specific
+<BlogPage: A Blog post>
+```
+
+When using `specific`, custom fields which only exist on the specific page (eg the body of a blog post) require an extra query to load. When using the `.specific` queryset method, these queries are done in bulk (one per page type), rather than one per instance.
+
+Instead, these extra fields can be "deferred". If only the base `Page` model's fields are needed, but you still need your instance's methods, the extra fields can be skipped when creating the specific instances:
+
+```python
+>>> Page.objects.all().specific(defer=True)
+[<BlogPage: A Blog post>, <BlogPage: Another Blog post>]
+>>> Page.objects.first()
+<Page: A Blog post>
+>>> Page.objects.first().specific_deferred
+<BlogPage: A Blog post>
+```
+
+The instances are the same - the difference comes when trying to access these extra fields:
+
+```python
+>>> page = Page.objects.first()
+>>> specific_page = page.specific  # A query is done here for the `body`
+>>> specific_deferred_page = page.specific_deferred  # No queries are done here
+>>> specific_deferred_page.body  # The query for `body` is done here
+```
+
+Because no additional queries are performed, and less data is operated on, "deferred" models are significantly faster than specific versions.
```

Comment:
```suggestion
Because no additional queries are performed, and less data is operated on, generating these "deferred" model instances can result in a significant boost in performance, particularly when operating on a large number of pages.
```